Adamant/Modules/Chat/View/ChatViewController.swift

```diff
@@ -713,6 +700,29 @@ private extension ChatViewController {
     func updateScrollDownButtonVisibility() {
         scrollDownButton.isHidden = isScrollPositionNearlyTheBottom
     }
+    
+    func updateDateHeaderIfNeeded() {
+        guard viewAppeared else { return }
+        
+        let targetY: CGFloat = targetYOffset + view.safeAreaInsets.top
+        guard let visibleIndexPaths = messagesCollectionView.indexPathsForVisibleItems.sorted(by: {
+            $0.row < $1.row
+        }) as [IndexPath]? else { return }
+        
+        for indexPath in visibleIndexPaths {
+            guard let cell = messagesCollectionView.cellForItem(at: indexPath)
```

Comment:
Do we need to proceed with the highest visible cell only?

Reply:
We go through the array and check if the cell matches our conditions